### PR TITLE
Flutter version auto increment 0.0.1

### PR DESCRIPTION
**Pull request by IbrahimTabba**
TagCheck

### What to do if the build fails?

At the moment contributors do not have access to the CI workflow triggered by StepLib PRs. In case of a failed build, we ask for your patience. Maintainers of Bitrise Steplib will sort it out for you or inform you if any further action is needed.

### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [x] __I will not move an already shared step version's tag to another commit__
- [x] I read the [Step Development Guideline](https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md)
- [x] I have a test for my Step, which can be run with `bitrise run test` (in the step's repository)
- [x] I did run `bitrise run audit-this-step` (in the step's repository - note, if you don't have this workflow in your `bitrise.yml`, [you can copy it from the step template](https://github.com/bitrise-steplib/step-template/blob/master/bitrise.yml).)
- [x] I read and accept the [Abandoned Step policy](https://github.com/bitrise-io/bitrise-steplib#abandoned-step-policy)

**New Step**
Thank you for the new Step share! The CI check might will fail due to our extended validation engine. Nothing to worry about yet, we will get back to you shortly.

### CODE REVIEW

File: steps/flutter-version-auto-increment/0.0.1/step.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+title: Flutter version auto increment
+summary: |
+  Increment current version and build number in pubspec.yaml file.
+description: |
+  Increment Project current version and build number file and rewite new value in pubspec.yaml file.
```
Comment (lpusok) [inline steps/flutter-version-auto-increment/0.0.1/step.yml:5]:
```suggestion
  Increment Project current version and build number file and write new value in pubspec.yaml file.
```

File: steps/flutter-version-auto-increment/0.0.1/step.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    is_expand: true
+    is_required: true
+    title: Version auto increment mode
+    value_options:
+    - patch
+    - minor
+    - major
+    - none
+  version_auto_increment_mode: patch
+- increment_build_number: "yes"
```
Comment (lpusok) [inline steps/flutter-version-auto-increment/0.0.1/step.yml:41]: Suggestion: To be consistent with iOS/Android versioning, could use Bitrise build number + a configurable offset (see https://devcenter.bitrise.io/en/builds/build-numbering-and-app-versioning.html#offsetting-the-build-version).

**Review by lpusok — CHANGES_REQUESTED** [review]
Thanks for the contribution.
Please update the title and the type in the description, then re-share the Step.

File: steps/flutter-version-auto-increment/0.0.1/step.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
+title: Flutter version auto increment
```
[the remark this answers is not in the document]
Comment (koral--) [reply steps/flutter-version-auto-increment/0.0.1/step.yml:1]: Both titles suggest that the subject is a version of Flutter itself, not a version of the project inside.

File: steps/flutter-version-auto-increment/0.0.1/step.yml (hunk 2 above)
Comment (IbrahimTabba) [reply steps/flutter-version-auto-increment/0.0.1/step.yml:41]: the source for both version and build number in flutter apps is the "version: X.X.X+X" stored in pubspec.yaml file.
this step is meant to be for production workflows, where the build number and version should be exactly one increment from the previous build, regardless of how many builds I did on bitrise and this variable BITRISE_BUILD_NUMBER

Comment (lpusok) [reply steps/flutter-version-auto-increment/0.0.1/step.yml:41]: Based on the [pubspec docs](https://dart.dev/tools/pub/pubspec#version) and [semver](https://semver.org/#spec-item-10), the build metadata (denoted by `+`) can be almost any string (git hash prefix for example) and it is not considered when comparing two versions.
My question is can we be sure that there is always a build number metadata commited? Incrementing a version without metadata (`1.2.3`) would not work. The same questoin also for a prerelease version being commited (`1.2.3-alpha`).

For reference checked an example package to understand conventions: https://pub.dev/packages/webview_flutter/versions.
Based on these, it seems that build number is usually not added, only for second build with the same minor, major and patch versions. Sometimes a prerelease version is used (`-alpha1` for example)

Comment (lpusok) [reply steps/flutter-version-auto-increment/0.0.1/step.yml:41]: An other idea: is this the input `increment_build_number` needed? It seems to me that `version_auto_increment_mode` option `none` could mean that we only increase build number. (Or add it if it was not yet added). 
The `none` option could even be renamed to `build_number_only` or similar.

**Review by lpusok — APPROVED** [review]

**Comment by lpusok** [comment]
@IbrahimTabba Should we release as is?

**Pull request closed without merging**